spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -383,14 +390,52 @@ public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
-    this.cacheEnabled = Boolean.parseBoolean(options.getOrDefault("cache-enabled", "true"));
-    Catalog catalog = buildIcebergCatalog(name, options);
+    this.cacheEnabled = PropertyUtil.propertyAsBoolean(options,
+        CatalogProperties.TABLE_CACHE_ENABLED, CatalogProperties.TABLE_CACHE_ENABLED_DEFAULT);
+
+    // If the user disabled caching and did not set the cache.expiration-interval-ms, we'll set it to zero for
+    // them on their behalf. If they disabled caching but explicitly set a non-zero cache expiration
+    // interval, we will fail initialization as that's an invalid configuration.
+    long defaultCacheExpirationInterval =
+        cacheEnabled ? CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS_DEFAULT : 0L;
+
+    this.cacheExpirationIntervalMs = PropertyUtil.propertyAsLong(options,
+        CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS,
+        defaultCacheExpirationInterval);
+
+    // Normalize usage of -1 to 0 as we'll call Duration.ofMillis on this value.
+    if (cacheExpirationIntervalMs < 0) {
+      this.cacheExpirationIntervalMs = 0;
+    }
+
+    Preconditions.checkArgument(cacheEnabled || cacheExpirationIntervalMs <= 0L,
+        "The catalog's table cache expiration interval must be set to zero via the property %s if caching is disabled",
+        CatalogProperties.TABLE_CACHE_EXPIRATION_INTERVAL_MS);
+
+    // If the user didn't specify TABLE_CACHE_EXPIRATION_INTERVAL_MS, put it into the SparkConf and the
+    // options map in case it's assumed to be elsewhere (such as cloning a spark session and
+    // re-instantiating the catalog).
```

Comment:
Not sure I understand what is happening here.

Reply:
Let's not set options. I think that removes the need for this then.

Reply:
Fair enough. Looking at it closer, I don't think it would be a huge issue.

My worry was that these options get pushed down, and are the options used to instantiate custom catalogs here: https://github.com/apache/iceberg/blob/58ceafb6c9e87aee7445141c283d0d2e06e38c5e/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L201-L235

We also modify the options map here: https://github.com/apache/iceberg/blob/58ceafb6c9e87aee7445141c283d0d2e06e38c5e/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L92-L106

But at least for Spark, it should be fine. If they're in the options map, they are (from user provided config), otherwise they're not but it shouldn't be an issue.

Reply:
I just checked and it shouldn't be an issue for Flink either. Will remove.

Reply:
This has been removed.